cirq-google/cirq_google/serialization/arg_func_langs.py

```diff
@@ -152,6 +152,16 @@ def arg_to_proto(
     ):
         # Some protobuf / numpy combinations do not support np.bool_, so cast.
         msg.arg_value.bool_values.values.extend([bool(x) for x in value])
+    elif isinstance(value, (list, tuple, np.ndarray)) and all(
+        isinstance(x, (int, np.integer)) for x in value
+    ):
+        msg.arg_value.int64_values.values.extend([int(x) for x in value])
+    elif isinstance(value, (list, tuple, np.ndarray)) and all(
+        isinstance(x, (float, np.floating)) for x in value
+    ):
+        msg.arg_value.double_values.values.extend([float(x) for x in value])
+    elif isinstance(value, (list, tuple, np.ndarray)) and all(isinstance(x, str) for x in value):
```

Comment:
Optional: It seems like a performance issue that we are iterating through the value list three times.  Are there any gates that are expecting large numpy arrays or anything that is expected to be a longer list.  If so, we may want to refactor this so we don't iterate through the list 3 times.

Reply:
changed to 2 loops the first finds the type/field and the second adds the values to the field.